contrib/slurm/slurmexecutor.py

```diff
@@ -0,0 +1,297 @@
+
+    with tempfile.TemporaryDirectory(dir=benchmark.config.scratchdir) as tempdir:
+
+        os.makedirs(os.path.join(tempdir, "upper"))
+        os.makedirs(os.path.join(tempdir, "work"))
+
+        tool_command = " ".join(args)
+        singularity_command = (
+            f"singularity exec "
+            f"-B $PWD:/lower --no-home "
+            f"-B {tempdir}:/overlay "
+            f'--fusemount "container:fuse-overlayfs -o lowerdir=/lower -o upperdir=/overlay/upper -o workdir=/overlay/work $HOME" '
+            f"{benchmark.config.singularity} {tool_command}"
+            if benchmark.config.singularity
+            else tool_command
+        )
+        srun_command = (
+            f"srun "
+            f"-t {srun_timelimit} "
+            f"-c {cpus} "
+            f"-o {log_file} "
+            f"--mem-per-cpu {mem_per_cpu} "
+            f"--threads-per-core=1 "
+            f"--ntasks=1 "
+            f"{singularity_command}"
+        )
+        jobid_command = (
+            f"{srun_command} 2>&1 | grep -o 'job [0-9]* queued' | grep -o '[0-9]*'"
+        )
+        seff_command = f"seff $({jobid_command})"
+        logging.debug("Command to run: %s", seff_command)
+        result = subprocess.run(
+            ["bash", "-c", seff_command], shell=False, stdout=subprocess.PIPE
+        )
+
+    # Runexec would populate the first 6 lines with metadata
+    with open(log_file, "r+") as file:
+        content = file.read()
+        file.seek(0, 0)
+        empty_lines = "\n" * 6
+        file.write(empty_lines + content)
+
+    exit_code, cpu_time, wall_time, memory_usage = parse_seff(str(result.stdout))
+
+    return {
+        "starttime": benchexec.util.read_local_time(),
+        "walltime": wall_time,
+        "cputime": cpu_time,
+        "memory": memory_usage,
+        "exitcode": ProcessExitCode(raw=exit_code, value=exit_code, signal=None),
+    }
+
+
+def parse_seff(result):
+    exit_code_pattern = re.compile(r"State: COMPLETED \(exit code (\d+)\)")
+    cpu_time_pattern = re.compile(r"CPU Utilized: (\d+):(\d+):(\d+)")
+    wall_time_pattern = re.compile(r"Job Wall-clock time: (\d+):(\d+):(\d+)")
+    memory_pattern = re.compile(r"Memory Utilized: (\d+\.\d+) MB")
+    exit_code_match = exit_code_pattern.search(result)
+    cpu_time_match = cpu_time_pattern.search(result)
+    wall_time_match = wall_time_pattern.search(result)
+    memory_match = memory_pattern.search(result)
+    exit_code = int(exit_code_match.group(1)) if exit_code_match else None
+    cpu_time = None
+    if cpu_time_match:
+        hours, minutes, seconds = map(int, cpu_time_match.groups())
+        cpu_time = hours * 3600 + minutes * 60 + seconds
+    wall_time = None
+    if wall_time_match:
+        hours, minutes, seconds = map(int, wall_time_match.groups())
+        wall_time = hours * 3600 + minutes * 60 + seconds
+    memory_usage = float(memory_match.group(1)) * 1000000 if memory_match else None
```

Comment:
Is the result produced by SLURM really in MB? Unfortunately, many tools wrongly report `MiB` values with the `MB` unit label, so it is better to double check what the value really means.

Reply:
Even though I could not find anything in the documentation, SLURM noticably uses 1000-fold increases in its prefixes `k`/`M`/`G` when parsing inputs, and I'd be very surprised if the measurements were in a different format than the resource allocation inputs.